File: Python/711A-bustoudayland.py
def print_bus(bus):
    for row in bus:
        print(''.join(row))


def solve():
    n = int(input())
    bus = [list(input()) for _ in range(n)]
    possible = False

    for i in range(n):
        for j in range(2):
            if bus[i][j*3] == 'O' and bus[i][j*3+1] == 'O':
                bus[i][j*3] = '+'
                bus[i][j*3+1] = '+'
                possible = True
                break
        if possible:
            break

    if not possible:
        print("NO")
    else:
        print("YES")
        print_bus(bus)


solve()

File: C++/A/711A-bustoudayland.cpp
#include <bits/stdc++.h>
using namespace std;
const int N = 1000;
char bus[N][5];
int main()
{
    int n;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < 5; j++)
        {
            cin >> bus[i][j];
        }
    }
    bool possible = false;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < 2; j++)
        {
            if (bus[i][j * 3] == 'O' && bus[i][j * 3 + 1] == 'O')
            {
                bus[i][j * 3] = '+';
                bus[i][j * 3 + 1] = '+';
                possible = true;
                break;
            }
        }
        if (possible)
            break;
    }
    if (possible)
    {
        cout << "YES\n";
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < 5; j++)
                cout << bus[i][j];
            cout << '\n';
        }
    }
    else
        cout << "NO";
}
